makelib/golang.mk

```diff
@@ -73,14 +78,17 @@ ifneq ($(GO_TEST_SUITE),)
 GO_TEST_FLAGS += -run '$(GO_TEST_SUITE)'
 endif
 
-GOPATH := $(shell go env GOPATH)
+# no GOPATH is set, since go modules are being used
+GOPATH := 
+
+# enable go modules
+export GO111MODULE=on
 
 # setup tools used during the build
-DEP_VERSION=v0.5.1
```

Comment:
Does this PR remove `dep` support entirely?

Reply:
Yes. Is there a reason to keep the support for `dep`? 
I initially created a separate file for `go modules` support. See the first commit.

Reply:
I don't think it should be a separate file necessarily, just would be nice to support `dep` as well for any projects that want to use it. Otherwise, upgrading the `build` submodule version will break their project.

Reply:
Modifying the existing `golang.mk`